toolchain/check/testdata/impl/no_prelude/name_poisoning.carbon

```diff
@@ -34,192 +18,13 @@ namespace N;
-// CHECK:STDERR: fn N.F1(x:! I);
-// CHECK:STDERR:             ^
-fn N.F1(x:! I);
-
-// CHECK:STDERR: fail_use_declaration_after_poison.carbon:[[@LINE+4]]:13: note: declared here [NameUseBeforeDeclNote]
-// CHECK:STDERR: interface N.I;
-// CHECK:STDERR:             ^
-// CHECK:STDERR:
-interface N.I;
-
-// CHECK:STDERR: fail_use_declaration_after_poison.carbon:[[@LINE+4]]:13: error: member name `I` not found in `N` [MemberNameNotFoundInScope]
-// CHECK:STDERR: fn N.F2(x:! N.I);
-// CHECK:STDERR:             ^~~
-// CHECK:STDERR:
-fn N.F2(x:! N.I);
-
-// --- fail_alias.carbon
-
-library "[[@TEST_NAME]]";
-
-interface I;
-namespace N;
-
-// CHECK:STDERR: fail_alias.carbon:[[@LINE+7]]:13: error: name `I` used before it was declared [NameUseBeforeDecl]
-// CHECK:STDERR: alias N.I = I;
-// CHECK:STDERR:             ^
-// CHECK:STDERR: fail_alias.carbon:[[@LINE+4]]:9: note: declared here [NameUseBeforeDeclNote]
-// CHECK:STDERR: alias N.I = I;
-// CHECK:STDERR:         ^
-// CHECK:STDERR:
-alias N.I = I;
-
-// --- fail_poison_multiple_scopes.carbon
-
-library "[[@TEST_NAME]]";
-
-interface I1;
-
-interface I2 {
-  interface I3 {
-    interface I4 {
-      // Use `package.I1` and poison:
-      // * `I2.I1`
-      // * `I2.I3.I1`
-      // * `I2.I3.I4.I1`
-      // CHECK:STDERR: fail_poison_multiple_scopes.carbon:[[@LINE+3]]:16: error: name `I1` used before it was declared [NameUseBeforeDecl]
-      // CHECK:STDERR:       fn F(x:! I1);
-      // CHECK:STDERR:                ^~
-      fn F(x:! I1);
-
-      // CHECK:STDERR: fail_poison_multiple_scopes.carbon:[[@LINE+7]]:17: note: declared here [NameUseBeforeDeclNote]
-      // CHECK:STDERR:       interface I1;
-      // CHECK:STDERR:                 ^~
-      // CHECK:STDERR:
-      // CHECK:STDERR: fail_poison_multiple_scopes.carbon:[[@LINE-6]]:16: error: name `I1` used before it was declared [NameUseBeforeDecl]
-      // CHECK:STDERR:       fn F(x:! I1);
-      // CHECK:STDERR:                ^~
-      interface I1;
-    }
-    // CHECK:STDERR: fail_poison_multiple_scopes.carbon:[[@LINE+7]]:15: note: declared here [NameUseBeforeDeclNote]
-    // CHECK:STDERR:     interface I1;
-    // CHECK:STDERR:               ^~
-    // CHECK:STDERR:
-    // CHECK:STDERR: fail_poison_multiple_scopes.carbon:[[@LINE-15]]:16: error: name `I1` used before it was declared [NameUseBeforeDecl]
-    // CHECK:STDERR:       fn F(x:! I1);
-    // CHECK:STDERR:                ^~
-    interface I1;
-  }
-  // CHECK:STDERR: fail_poison_multiple_scopes.carbon:[[@LINE+4]]:13: note: declared here [NameUseBeforeDeclNote]
-  // CHECK:STDERR:   interface I1;
-  // CHECK:STDERR:             ^~
-  // CHECK:STDERR:
-  interface I1;
-}
-
-// --- ignored_poison_in_import.carbon
-
-library "[[@TEST_NAME]]";
-import library "poison";
-
-// This doesn't fail.
-interface N.I;
-
 // --- poison.impl.carbon
 
 impl library "[[@TEST_NAME]]";
 
 // TODO: #4622 This should fail since `N.I` was poisoned in the api.
 interface N.I;
```

Comment:
Should this have been moved because it's only testing interface logic? The interface has no `impl` here.

Reply:
Done.